### ISSUE TITLE
ConnectionType(relay.Node) incorrectly prohibits connections to Node.

### ISSUE THREAD

**Issue by erydo** [graphql-python/graphene#233]
The connection types are filled with assertions that the target type is a non-abstract subclass of `Node` ("`ObjectTypes in a edge have to be Nodes`"). But `Node` itself *should* be perfectly valid. It allows you to create general connections to multiple `Node` subtypes which can be expanded with fragments.

For example:

```graphql
type Foo implements Node {
  id: ID!
  myFoo: String!
}
type Bar implements Node {
  id: ID!
  myBar: String!
}
type NodeEdge {
  node: Node
  cursor: String!
}
type NodeConnection {
  pageInfo: PageInfo!
  edges: [NodeEdge]
}
type Viewer {
  foosAndBars: NodeConnection
}
type Query {
  node(id: ID!): Node
  viewer: Viewer!
}
```

Queried via:
```graphql
query {
  viewer {
    myFoosAndBars {
      edges {
        node {
          id
          ...on Foo {
            myFoo
          }
          ...on Bar {
            myBar
          }
        }
      }
    }
  }
}
```

**Comment by erydo** [comment]
Additionally, from the [Relay Connection specification](https://facebook.github.io/relay/graphql/connections.htm#sec-Node), which supports that this should be allowed:

>An “Edge Type” must contain a field called node. This field must return either a **Scalar, Enum, Object, Interface, Union, or a Non‐Null wrapper around one of those types.** Notably, this field cannot return a list.
> 
> (…) Relay can perform certain optimizations if this field returns an object that implements Node, however, this is not a strict requirement for use of Relay.

**Comment by Globegitter** [comment]
@erydo https://github.com/graphql-python/graphene/pull/237 should fix this :)

**Comment by syrusakbary** [comment]
This should be fixed in `1.0.dev` (master). Closing the issue.
